src/core/document.js

```diff
@@ -827,6 +828,53 @@ class PDFDocument {
     return this.xfaFactory !== null;
   }
 
+  loadXfaFonts(handler, task) {
+    const resources = this.catalog.acroForm.get("DR");
+    const fonts = resources.get("Font");
+    if (!(fonts instanceof Dict)) {
+      return Promise.resolve(null);
+    }
+
+    const objectLoader = new ObjectLoader(resources, ["Font"], this.xref);
+    const partialEvaluator = new PartialEvaluator({
+      xref: this.xref,
+      handler,
+      pageIndex: -1,
+      idFactory: this._globalIdFactory,
+      fontCache: this.catalog.fontCache,
+      builtInCMapCache: this.catalog.builtInCMapCache,
+    });
+    const operatorList = new OperatorList();
+    const initialState = {
+      font: null,
+      clone() {
+        return this;
+      },
+    };
+
+    return objectLoader.load().then(() => {
+      fonts.forEach((pdfFontName, font) => {
+        const descriptor = font.get("FontDescriptor");
+        if (descriptor instanceof Dict) {
+          const fontFamily = descriptor.get("FontFamily");
+          const fontWeight = descriptor.get("FontWeight");
+          const italicAngle = descriptor.get("ItalicAngle");
+
+          partialEvaluator.handleSetFont(
+            resources,
+            [Name.get(pdfFontName), 1],
+            null,
+            operatorList,
+            task,
+            initialState,
+            /* fallbackFontDict = */ null,
+            /* cssFontInfo = */ { fontFamily, fontWeight, italicAngle }
+          );
+        }
+      });
+    });
+  }
+
```

Comment:
This method is doing a bunch of *synchronous* data lookups, all of which can throw `MissingDataException`s that will break PDF document loading.
Since this method is intended to be called asynchronously, it really needs to guard all of it's data accesses properly and below I'm suggesting a few (untested) changes to prevent issues later on.
```suggestion
  async loadXfaFonts(handler, task) {
    // (Remove this comment) While this is currently *indirectly* guarded by where you're calling the `loadXfaFonts` method from, that's a brittle approach which we should avoid depending on.
    const acroForm = await this.pdfManager.ensureCatalog("acroForm");
    
    const resources = await acroForm.getAsync("DR");
    if (!(resources instanceof Dict)) {
      return;
    }
    const objectLoader = new ObjectLoader(resources, ["Font"], this.xref);
    await objectLoader.load();
    
    const fontRes = resources.get("Font");
    if (!(fontRes instanceof Dict)) {
      return;
    }

    const partialEvaluator = new PartialEvaluator({
      xref: this.xref,
      handler,
      pageIndex: -1,
      idFactory: this._globalIdFactory,
      fontCache: this.catalog.fontCache,
      builtInCMapCache: this.catalog.builtInCMapCache,
    });
    const operatorList = new OperatorList();
    const initialState = {
      font: null,
      clone() {
        return this;
      },
    };

    const fonts = new Map();
    fontRes.forEach((fontName, font) => {
      fonts.set(fontName, font);
    });
    const promises = [];
    
    for ([fontName, font] of fonts) {
      const descriptor = font.get("FontDescriptor");
      if (descriptor instanceof Dict) {
        const fontFamily = descriptor.get("FontFamily");
        const fontWeight = descriptor.get("FontWeight");
        const italicAngle = descriptor.get("ItalicAngle");

        // (Remove this comment) You also need to be able to wait for all fonts to be loaded before returning.
        const promise = partialEvaluator.handleSetFont(
          resources,
          [Name.get(fontName), 1],
          /* fontRef = */ null,
          operatorList,
          task,
          initialState,
          /* fallbackFontDict = */ null,
          /* cssFontInfo = */ { fontFamily, fontWeight, italicAngle }
        );
        // (Remove this comment) If necessary, you could do `promises.push(promise.catch(() => {});` here to prevent one bad font from breaking all font loading.
        promises.push(promise);
      }
    });
    await Promise.all(promises);
  }

```